x-pack/solutions/observability/plugins/profiling/public/views/functions/differential_topn/index.tsx

```diff
@@ -147,6 +149,21 @@ export function DifferentialTopNFunctionsView() {
     });
   }
 
+  useEffect(() => {
+    if (state.status === AsyncStatus.Settled || comparisonState.status === AsyncStatus.Settled) {
+      onPageReady({
+        meta: {
+          rangeFrom,
+          rangeTo,
+        },
+        customMetrics: {
+          key1: 'totalCount',
+          value1: state.data?.TotalCount ?? 0,
+        },
+      });
+    }
+  }, [state, comparisonState, onPageReady, rangeTo, rangeFrom]);
```

Comment:
We probably just want `state.status` instead of the entire `state` to be in the dependency list. If another complex object is part of `state`, it will inadvertently cause the `useEffect` to be triggered

```suggestion
  }, [state.status, comparisonState, onPageReady, rangeTo, rangeFrom]);
```

Reply:
that would probably explain the re-renders I was seeing